tests/index.ts

```diff
@@ -0,0 +1,165 @@
+  const testTargets = new Map<string, ITestTarget>();
+  // Sort guarantees basic.in comes before basic.out
+  fs.readdirSync(dirname)
+    .sort()
+    .forEach((file: string) => {
+      const inFileRegex = /(?<prefix>.+)\.in\.(?<extension>.+)$/;
+      const foundIn = file.match(inFileRegex);
+      if (foundIn) {
+        const prefix = foundIn.groups?.prefix;
+        if (prefix && (namedTestPrefixes.includes(prefix) || namedTestPrefixes.length == 0)) {
+          testTargets.set(prefix, {
+            prefix,
+            inputPath: path.join(parentTestDirName, file),
+            outputPath: "",
+          });
+        }
+        return;
+      }
+
+      const outFileRegex = /(?<prefix>.+)\.out\.(?<extension>.+)$/;
+      const foundOut = file.match(outFileRegex);
+      if (foundOut) {
+        const prefix = foundOut.groups?.prefix;
+        if (prefix) {
+          const maybe_target = testTargets.get(prefix);
+          if (maybe_target) {
+            maybe_target.outputPath = path.join(parentTestDirName, file);
+          }
+        }
+      }
+    });
+  return Array.from(testTargets.values()).filter((target) => target.outputPath.length > 0);
+};
+
+// Setup the TrunkDriver to run tests in a 'dirname'. 'dirname' should be the path to the test subdirectory in a linter folder
+// If a 'linterName' is specified, that linter will be enabled during setup according to specification in 'inputArgs'.
+export const setupDriver = (
+  dirname: string,
+  inputArgs: ITestingArguments,
+  setupSettings: ISetupSettings,
+  linterName?: string
+): TrunkDriver => {
+  const driver = new TrunkDriver(dirname, inputArgs, setupSettings, linterName);
+
+  beforeAll(() => {
+    driver.SetUp();
+  });
+
+  afterAll(() => {
+    driver.TearDown();
+  });
+  return driver;
+};
+
+// Test that running a linter filtered by 'linterName' on the test files in 'dirname' produces the desired output json.
+// Either detect input and ouptut files automatically, or specify their prefixes as `namedTestPrefixes`. Output files should be of
+// .json type, regardless of the input file format.
+export const defaultLinterCheckTest = (
+  dirname: string,
+  linterName: string,
+  namedTestPrefixes: string[] = []
+) => {
+  describe(`Testing linter ${linterName}`, () => {
+    // Step 1: Parse any custom inputs
+    const inputArgs = parseInputs();
+
+    // Step 2: Define test setup and teardown
+    const driver = setupDriver(dirname, inputArgs, {}, linterName);
+
+    // Step 3: Detect test files to run
+    const linterTestTargets = detectTestTargets(dirname, namedTestPrefixes);
+
+    // Step 4: Asynchronously run each test
+    linterTestTargets.forEach((test_target) => {
+      it(test_target.prefix, async () => {
+        const test_run_result = await driver.RunCheckUnit(test_target.inputPath, linterName);
+        expect(test_run_result.success);
+
+        const expected_out_file = path.join(dirname, path.parse(test_target.outputPath).base);
```

Comment:
nit: I recommend destructuring `test_target` to make things a bit more terse

```suggestion
    linterTestTargets.forEach(({ prefix, inputPath, outputPath }) => {
      it(prefix, async () => {
        const test_run_result = await driver.RunCheckUnit(inputPath, linterName);
        expect(test_run_result.success);

        const expected_out_file = path.join(dirname, path.parse(outputPath).base);
```